tests/_workflow_exec.py

```
@@ -0,0 +1,199 @@
+        "GITHUB_OUTPUT": str(github_output),
+        "GITHUB_WORKSPACE": str(workspace),
+        "RUNNER_TEMP": str(workspace / "_runner_temp"),
+    }
+    (workspace / "_runner_temp").mkdir(exist_ok=True)
+    # The step's own declared env, with unresolved ${{ }} expressions left to
+    # the caller to substitute — a test that forgets is passing a literal
+    # expression, which is visible rather than silently empty.
+    step_env.update({k: str(v) for k, v in (step.get("env") or {}).items()})
+    step_env.update(env or {})
+
+    proc = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)